pokemongo_bot/cell_workers/evolve_all_worker.py

```diff
@@ -53,17 +53,19 @@ def work(self):
 
     def _should_run(self):
         # Will skip evolving if user wants to use an egg and there is none
-        skip_evolves = False
 
-        if self.config.evolve_all:
-            return skip_evolves
+        if not self.config.evolve_all:
+            return False
 
         # Pop lucky egg before evolving to maximize xp gain
         use_lucky_egg = self.config.use_lucky_egg
         lucky_egg_count = self.bot.item_inventory_count(Item.ITEM_LUCKY_EGG.value)
 
-        if use_lucky_egg and lucky_egg_count > 0:
-            logger.log('Using lucky egg ... you have {}'
+        if not use_lucky_egg: # If not using lucky eggs, skipping checks below
+            return True
+
+        if lucky_egg_count > 0:
```

Comment:
so every 30 mins it will use a lucky egg.  This doesn't seem ideal.  Not only that, currently it will send the API request every second (technically tick) which is not good at all.